code/triangle/ViewModel/src/main/java/ru/unn/agile/triangle/viewmodel/ViewModel.java

```diff
@@ -85,11 +103,61 @@ public void calculate() {
             Triangle triangle = new Triangle(a, b, c);
             result.set(operation.get().apply(triangle).toString());
             status.set(Status.SUCCESS.toString());
+            StringBuilder message = new StringBuilder(LogMessages.CALCULATE_WAS_PRESSED);
+            message.append("Arguments: ax = ").append(ax.get())
```

Comment:
These 107-112 strings are practically the same as 413-419 and 141-147 and so on....
It will be better to use just a string constant template and format function to fill the gaps in it,
instead of creating it everytime, everywhere.
constant = "Arguments: ax = %s;  ay = %s.............................."
String.format(constant, ax, ay, .....);